src/NHibernate/SqlCommand/SqlSelectBuilder.cs

```diff
@@ -110,6 +113,9 @@ public SqlSelectBuilder SetOuterJoins(SqlString outerJoinsAfterFrom, SqlString o
 			}
 
 			this.outerJoinsAfterWhere = tmpOuterJoinsAfterWhere;
+
+			hasOuterJoin = hasOuterJoin || 
+			               outerJoinsAfterFrom != null && outerJoinsAfterFrom.ToString().ToLowerInvariant().Contains("outer join");
```

Comment:
The test should be moved into `GetForUpdateString`, and done only if the dialect does not support for update on outer joins. It would avoid doing such a string scan for all other cases.

By the way, is it really required here to test the `outerJoinsAfterFrom` contains an outer join? Shouldn't it be enough to test the string is not null or just whitespaces?